### PR TITLE
React env refactor

### PR DESCRIPTION
**Pull request by benjgil**
## Proposed Changes

- enable overriding of react-app deps 
- this enables e.g.  updating the version of react used by the app dev server
-

### CODE REVIEW

File: scopes/react/react/react.env.ts [hunk 1]
```diff
@@ -67,7 +67,7 @@ const prettierConfig = require('./prettier/prettier.config.js');
 type CompilerMode = 'build' | 'dev';
 
 type GetBuildPipeModifiers = PipeServiceModifiersMap & {
-  tsModifier?: PipeServiceModifier;
+  compilerModifier?: PipeServiceModifier;
```
Comment (GiladShoham) [inline scopes/react/react/react.env.ts:70]: This is a breaking change, you can't just change it. it will break many users
also, I really want to remove this soon, getBuildPipe shouldn't get transformers (like the other get*** that shouldn't get them). 
let's undo this change for now, and discuss everything around build pipeline seperately

File: scopes/react/react/apps/web/react-app-options.ts [hunk 2]
```diff
@@ -119,4 +120,10 @@ export type ReactAppOptions = {
    * favicon for the app. You can pass an abs path (using require.resolve()) or a url.
    */
   favicon?: string;
+
+  /**
+   * dependencies to be configured for the application - e.g. version of react
+   * items here will override any defaults that are already configured for a package+dependencyType
+   */
+  dependencies?: WorkspacePolicy;
```
Comment (GiladShoham) [inline scopes/react/react/apps/web/react-app-options.ts:128]: please move all these dependencies changes into a different PR
I want to review it more carefully, and I'm going to touch the exact same area just now, so we should be in sync on this.
Keeping this inside this PR will block me from merging all the other refactor / renames changes.

**Review by GiladShoham — CHANGES_REQUESTED** [review]

File: scopes/react/react/react.env.ts (hunk 1 above)
Comment (benjgil) [reply scopes/react/react/react.env.ts:70]: I dont think it is a breaking change, it's local to the env and only used here

File: scopes/react/react/apps/web/react-app-options.ts (hunk 2 above)
Comment (benjgil) [reply scopes/react/react/apps/web/react-app-options.ts:128]: it's the only way to make the env overridable for the react17/18/etc apps to not have to create a new app-type

I'll move it to another PR but this will block the react18 env...

File: scopes/react/react/react.env.ts (hunk 1 above)
Comment (benjgil) [reply scopes/react/react/react.env.ts:70]: And i agree, I have no idea why it gets transformers and where it gets them from...

Comment (benjgil) [reply scopes/react/react/react.env.ts:70]: (when i say it's only used here, it's only used by the UseTypescript function in main.runtime, which has also been updated)
